src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,57 @@
+@import "../utils/variables.scss";
+@import "../utils/mixins.scss";
+
+%day-style {
+  background-color: #eee;
+  border: 1px solid black;
+}
+
+%display-center {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  @extend %display-center;
+  justify-content: flex-start;
+  flex-wrap: wrap;
+  gap: $calendar-day-gap;
+  max-width: $calendar-width;
+  padding: $calendar-day-padding-y $calendar-day-padding-x;
+
+  @each $day, $number in $days-of-week {
+    &--start-day-#{$day} :nth-child(1) {
+      margin-left: $number * $calendar-day-width + $number;
+    }
+  }
+
+  &__day {
+    display: none;
+    @extend %day-style;
+    @include square(100px);
+    transition: translate 0.5s, background-color 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      translate: 0 -20px;
+    }
+
+    @for $days-number from 1 through 31 {
+      &--#{$days-number}::before {
+        content: "#{$days-number}";
+        font-family: Arial, sans-serif;
+        font-size: 30px;
+        height: 100%;
+        @extend %display-center;
+      }
+    }
+  }
+
+  @for $month-length from 28 through 31 {
+    &--month-length-#{$month-length} :nth-child(-n + #{$month-length}) {
+      display: block;
+    }
+  }
```

Comment:
maybe it should be somewhere upper in code, in 29 line, If this doesn't break functionality,